app/src/main/java/pm/gnosis/heimdall/ui/transactions/details/AssetTransferTransactionDetailsFragment.kt

```diff
@@ -0,0 +1,248 @@
+import pm.gnosis.heimdall.common.utils.Result
+import pm.gnosis.heimdall.common.utils.snackbar
+import pm.gnosis.heimdall.data.repositories.models.ERC20TokenWithBalance
+import pm.gnosis.heimdall.data.repositories.models.Safe
+import pm.gnosis.heimdall.ui.exceptions.LocalizedException
+import pm.gnosis.models.Transaction
+import pm.gnosis.models.TransactionParcelable
+import pm.gnosis.utils.asEthereumAddressStringOrNull
+import pm.gnosis.utils.hexAsBigIntegerOrNull
+import pm.gnosis.utils.nullOnThrow
+import pm.gnosis.utils.stringWithNoTrailingZeroes
+import timber.log.Timber
+import java.math.BigInteger
+import javax.inject.Inject
+
+
+class AssetTransferTransactionDetailsFragment : BaseTransactionDetailsFragment() {
+
+    @Inject
+    lateinit var subViewModel: AssetTransferTransactionDetailsContract
+
+    private val adapter by lazy {
+        // Adapter should only be created if we need it
+        TokensSpinnerAdapter(context!!)
+    }
+    private val safeSubject = BehaviorSubject.createDefault<Optional<BigInteger>>(None)
+    private val inputSubject = PublishSubject.create<AssetTransferTransactionDetailsContract.CombinedRawInput>()
+    private var editable: Boolean = false
+    private var originalTransaction: Transaction? = null
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View?
+            = inflater.inflate(R.layout.layout_transaction_details_asset_transfer, container, false)
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        editable = arguments?.getBoolean(ARG_EDITABLE, false) ?: false
+        val transaction = arguments?.getParcelable<TransactionParcelable>(ARG_TRANSACTION)?.transaction
+        originalTransaction = transaction
+        layout_transaction_details_asset_transfer_max_amount_button.visibility = if (editable) View.VISIBLE else View.GONE
+        layout_transaction_details_asset_transfer_divider_qr_code.visibility = if (editable) View.VISIBLE else View.GONE
+        toggleTransactionInput(editable)
+    }
+
+    override fun onStart() {
+        super.onStart()
+
+        val safe = arguments?.getString(ARG_SAFE)?.hexAsBigIntegerOrNull()
+        setupSafeSpinner(layout_transaction_details_asset_transfer_safe_input, safe)
+        layout_transaction_details_asset_transfer_token_input.adapter = adapter
+
+        disposables += subViewModel.loadFormData(originalTransaction)
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(::setupForm, Timber::e)
+    }
+
+    private fun setupForm(info: AssetTransferTransactionDetailsContract.FormData) {
+        layout_transaction_details_asset_transfer_to_input.setText(info.to?.asEthereumAddressStringOrNull())
+        layout_transaction_details_asset_transfer_amount_input.setText(info.tokenAmount?.let { info.token?.convertAmount(it)?.stripTrailingZeros()?.toPlainString() })
```

Comment:
`stripTrailingZeros()?.toPlainString()` can be replaced with extension function `stringWithNoTrailingZeroes`